### DISCUSSION TITLE
Using Images from post directory

### DISCUSSION THREAD

**Discussion by yshalsager** [razonyang/hugo-theme-bootstrap#456] [Q&A] [answered]
So, I have been finally able to update the theme to the latest release, including Hugo pipes version, but unfortunately this commit https://github.com/razonyang/hugo-theme-bootstrap/commit/c08bca72ebc43319d04f419c66f7f88e97755d97 is breaking all posts images preview on my website because of my posts structure which is:

```
post1
├── images
│   └── kimberly-farmer-lUaaKCUANVI-unsplash.jpg
└── index.md
```
Inside `index.md` there's the image attribute in the YAML metadata like `images = ["images/kimberly-farmer-lUaaKCUANVI-unsplash.jpg"]`

I don't like putting all images in one directory and prefer my current approach.

I have tried debugging `.Params.images` and it points to `/images/kimberly-farmer-lUaaKCUANVI-unsplash.jpg` indeed, but this URL should be preceded by post URL in order for the image to be shown correctly.

Suggestions will be appreciated!

CC: @razonyang @ElanHasson

**Comment by razonyang** [comment, marked as answer]
Sorry for the inconvenience. Because the previous approach is in conflict with the Hugo built in Twitter Card and Open graph templates(the `images` will not be treated as page resources).

If you're using a page resource as the thumbnails. You'll need to move the thumbnail image to page level, and rename it to `feature.png`, `cover.png` or `thumbnail.png` etc. Finally, delete the `images` parameter.

For example: `/posts/my-page/images/my-thumbnail.png` to `/posts/my-page/featured.png`.

As for the rest of images under `images` folder are still processed by Markdown correctly.

**Reply by yshalsager** [reply]
Thanks @razonyang, I will work on this.
I have noticed that in search page, the thumbnail image from yaml is still used:
![image](https://user-images.githubusercontent.com/17043808/159274374-d87b9556-2729-4ec2-a26e-bc11203efe08.png)
And updating the image to thumbnail.png in post root directory as you suggested broke the thumbnail in search page

**Reply by razonyang** [reply]
Thanks for pointing this issue out.

**Reply by yshalsager** [reply]
@razonyang Welcome!
Another thing that has been broken as well since I updated the theme is: homepage shows notes from docs layout, while before it didn't and only showed notes with post layout (and this is what I want to)

My current content structure is:

```
content
├── about
├── archives
├── categories
├── contact
├── notes
│   └── books
│       └── xyz
├── posts
│   ├── post
```
Inside notes parent directory, there's an `_index.md` with required meta as mentioned in docs, as well as every invidual book note markdown file:
```
type = "docs"
layout = "docs"
```

**Reply by razonyang** [reply]
What's the value of `mainSections`?

**Reply by razonyang** [reply]
@yshalsager FYI, The thumbnails issue on the search page should be fixed by #459.
Preview: https://deploy-preview-459--hugo-theme-bootstrap.netlify.app/en/search/?q=markdown

**Reply by yshalsager** [reply]
> What's the value of `mainSections`?

Ah I was missing it, adding it fixed the problem!

**Reply by yshalsager** [reply]
> @yshalsager FYI, The thumbnails issue on the search page should be fixed by #459. Preview: [deploy-preview-459--hugo-theme-bootstrap.netlify.app/en/search/?q=markdown](https://deploy-preview-459--hugo-theme-bootstrap.netlify.app/en/search/?q=markdown)

Amazing! Thank you so much :)
Time to deploy the updated site ⚙️

**Reply by razonyang** [reply]
@yshalsager My pleasure.

**Comment by ElanHasson** [comment]
@razonyang a few folks have run into this issue-- what can be done to improve the change I introduced or would you consider it to be a documentation "problem"?

**Reply by razonyang** [reply]
Done 0f518d1. It should be more visible than [Thumbnail Selection Order](https://hbs.razonyang.com/en/docs/content/#thumbnail-selection-order).
